ktor-server/ktor-server-test-base/common/src/io/ktor/server/test/base/BaseTest.kt

```diff
@@ -4,13 +4,33 @@
 
 package io.ktor.server.test.base
 
-import kotlinx.coroutines.*
-import kotlinx.coroutines.test.*
-import kotlin.time.*
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.test.TestResult
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.seconds
 
 expect abstract class BaseTest() {
     open val timeout: Duration
+
+    open fun beforeTest()
+    open fun afterTest()
+
     fun collectUnhandledException(error: Throwable) // TODO: better name?
-    fun runTest(block: suspend CoroutineScope.() -> Unit): TestResult
-    fun runTest(timeout: Duration, block: suspend CoroutineScope.() -> Unit): TestResult
+    fun runTest(timeout: Duration = 60.seconds, block: suspend CoroutineScope.() -> Unit): TestResult
```

Comment:
I'm not sure about such a big default timeout. Successful tests usually run faster.

Reply:
Could you tell me if you have any value in mind?

Reply:
According to [our Develocity statistics](https://ge.jetbrains.com/scans/tests?search.rootProjectNames=ktor&tests.container=*.server.*&tests.sortField=MEAN_DURATION), 30 seconds should be enough for all tests.